src/main/java/com/sreeram/bccleaderboard/services/LichessService.java

```diff
@@ -32,34 +34,55 @@ public class LichessService implements IService {
 
   @Override
   public LeaderboardResponse getLeaderboard() {
+    Map<String, List<PlayerTournamentOutcome>> metrics = new HashMap<>();
     List<Tournament> tournaments = getCurrentMonthFinishedTournaments();
-    HashMap<String, Player> topPlayers = getTopPlayersOfTheMonth(tournaments);
-    return new LeaderboardResponse(topPlayers);
+    tournaments.forEach(t -> {
+      computeAndUpdatePlayerLevelMetrics(metrics, getTournamentTopTenResults(t));
+    });
+    List<Player> players = mapMetricsToPlayerData(metrics);
+    return new LeaderboardResponse(players);
   }
 
-  public HashMap<String, Player> getTopPlayersOfTheMonth(List<Tournament> tournaments) {
-    HashMap<String, Player> results = new HashMap<>();
-    for (Tournament tmt : tournaments) {
-      List<TournamentPlayerResult> top10 = getClient().getTopTenPlayers(tmt);
-      if(top10 == null) {
-        LOGGER.warn("Top ten players list from tournament {} is null" , tmt);
-        continue;
-      }
-      int listSize = top10.size();
-      int score = 0;
-      for (TournamentPlayerResult playerResult: top10) {
-        String username = playerResult.getUsername();
-        Player player = results.getOrDefault(username, new Player(username));
-        int clubScore = listSize - score;
-        // TODO - Update club score according to the points and tiebreak scored in the tournamentPlayerResult
-        player.addTournamentResults(playerResult, clubScore);
-        results.put(username, player);
-        score++;
+  private void computeAndUpdatePlayerLevelMetrics(Map<String, List<PlayerTournamentOutcome>> metrics, List<TournamentPlayerResult> topTenResults) {
+    TournamentPlayerResult previousPlayerResult = null;
+    int previousClubScore = 10, rank = 1;
+    int listSize = topTenResults.size(); 
+    for (TournamentPlayerResult r: topTenResults) {
+      int clubScore = listSize - rank + 1;
+      if (previousPlayerResult != null) {
+        if (previousPlayerResult.getPoints().equals(r.getPoints()) &&
+            previousPlayerResult.getTiebreak().equals(r.getTiebreak())) {
+              clubScore = previousClubScore;
+            } 
       }
+      updatePlayerWiseMetrics(metrics, r, clubScore);
+      previousPlayerResult = r;
+      previousClubScore = clubScore;
```

Comment:
Good point. But practically there will definitely be 10+.
But yes, 321 is wrong.10,9,8 should be considered actually. Because people who participate in tournaments with < 10 participants will be getting low scores.

Reply:
Change made